lib/logstash/filters/translate.rb

```diff
@@ -157,39 +207,17 @@ def filter(event)
         @logger.info("refreshing dictionary file")
       end
     end
+
+    #if destination is not set default to #{@field}_translation
+    @destination ||= "#{@field}_translation"
     
     return unless event.include?(@field) # Skip translation in case event does not have @event field.
-    return if event.include?(@destination) and not @override # Skip translation in case @destination field already exists and @override is disabled.
+    return if event.include?(@destination) &&  !@override # Skip translation in case @destination field already exists and @override is disabled.
 
     begin
-      #If source field is array use first value and make sure source value is string
-      source = event[@field].is_a?(Array) ? event[@field].first.to_s : event[@field].to_s
-      matched = false
-      if @exact
-        if @regex
-          key = @dictionary.keys.detect{|k| source.match(Regexp.new(k))}
-          if key
-            event[@destination] = @dictionary[key]
-            matched = true
-          end
-        elsif @dictionary.include?(source)
-          event[@destination] = @dictionary[source]
-          matched = true
-        end
-      else 
-        translation = source.gsub(Regexp.union(@dictionary.keys), @dictionary)
-        if source != translation
-          event[@destination] = translation.force_encoding(Encoding::UTF_8)
-          matched = true
-        end
-      end
+      determine_field_type(event)
```

Comment:
this is now a uber method that encapsulates everything from analyzing the source, doing the lookup, changing the event and calling filter_matched
I suggest either moving back the code from `determine_field_type` back here into `filter` or somehow split it into two or three methods but that seems harder.